lib/metanorma/collection/collection.rb

```diff
@@ -0,0 +1,228 @@
+# frozen_string_literal: true
+
+require "relaton"
+require "relaton/cli"
+require "metanorma-utils"
+require_relative "../util/util"
+require_relative "../util/disambig_files"
+require_relative "../collectionconfig/collectionconfig"
+
+module Metanorma
+  class FileNotFoundException < StandardError; end
+
+  class AdocFileNotFoundException < StandardError; end
+
+  # Metanorma collection of documents
+  class Collection
+    attr_reader :file
+
+    # @return [Array<String>] documents-inline to inject the XML into
+    #   the collection manifest; documents-external to keeps them outside
+    attr_accessor :directives, :documents, :bibdatas, :coverpage, :dirname
+    attr_accessor :disambig, :manifest, :bibdata, :compile
+
+    # @param file [String] path to source file
+    # @param dirname [String] directory of source file
+    # @param directives [Array<String>] documents-inline to inject the XML into
+    #   the collection manifest; documents-external to keeps them outside
+    # @param bibdata [RelatonBib::BibliographicItem]
+    # @param manifest [Metanorma::CollectionManifest]
+    # @param documents [Hash<String, Metanorma::Document>]
+    # @param prefatory [String]
+    # @param coverpage [String]
+    # @param final [String]
+    def initialize(**args)
```

Comment:
Let's not use `**args` but actually write out what args you are expecting and what defaults to empty.